SlackBot/Program.cs

```diff
@@ -403,4 +463,90 @@ private static async void SendMessage(string textMessage, string messageTimeStam
     }
     #endregion
   }
+  static class GetThreadExtension
+  {
+    /// <summary>
+    /// Получить весь тред.
+    /// </summary>
+    /// <param name="messageTimestamp">Время первого сообщения треда.</param>
+    /// <param name="channel">Канал треда.</param>
+    /// <returns>Все сообщения треда.</returns>
+    public static ThreadDTO GetThread(this IConversationsApi conversations, string messageTimestamp, string channel)
+    {
+      var messages = conversations.Replies(channel, messageTimestamp, limit: 50).Result.Messages;
+      List<MessageDTO> messageDTO = new List<MessageDTO>(messages.Count);
+      ThreadDTO thread = new ThreadDTO();
+      foreach (var message in messages)
+      {
+        messageDTO.Add(new MessageDTO
+        {
+          Text = message.Text,
+          Ts = message.Ts,
+          User = message.User,
+          Files = GetFiles(new List<FileDTO>())
+        });
+
+        List<FileDTO> GetFiles(List<FileDTO> list)
+        {
+          if (message.Files.Count == 0) return null;
+          foreach (var file in message.Files)
+          {
+            list.Add(new FileDTO
+            {
+              Name = file.Name,
+              UrlPrivateDownload = file.UrlPrivateDownload
+            });
+          }
+          return list;
+        }
+      }
+      thread.Messages = messageDTO;
+      return thread;
+    }
+
+
+  }
+
+  public static class PostEphemeralMessageToUserExtension
+  {
+    /// <summary>
+    /// Отправляет пользователю сообщение типа Only visible to you.
+    /// </summary>
+    /// <param name="message">Сообщение для отправки.</param>
+    /// <param name="userId">Id пользователя, которому нужно отправить сообщение.</param>
+    /// <param name="channelId">Id чата для отправки сообщения.</param>
+    public static void PostEphemeralMessageToUser(this IChatApi chat, string text, string userId, string channelId)
+    {
+      Message message = new Message
+      {
+        Text = text,
+        Channel = channelId
+      };
+      chat.PostEphemeral(userId, message);
+    }
+  }
+
+  static class GetUserNameByIdExtension
+  {
+    /// <summary>
+    /// Получить ник пользователя по его id.
+    /// </summary>
+    /// <param name="userId"></param>
+    /// <returns>Ник пользователя.</returns>
+    public static UserDTO GetUserNameById(this IUsersApi user, string userId)
+    {
+      var name = user.Info(userId).Result.Name;
+      var realName = user.Info(userId).Result.RealName;
+      DTOs.User DTOsUser = new DTOs.User
+      {
+        Name = name,
+        RealName = realName
+      };

```

Comment:
```suggestion
    public static UserDTO GetUserNameById(this IUsersApi users, string userId)
    {
      var user = users.Info(userId).Result;
      DTOs.User DTOsUser = new DTOs.User
      {
        Name = user.Name,
        RealName = user.RealName
      };
```
Давай так. Чтобы не делать двойной вызов `Info(userId)`.